src/zeek.bif

```diff
@@ -1895,12 +1896,33 @@ function reading_live_traffic%(%): bool
 ##
 ## Returns: True if reading traffic from a network trace.
 ##
-## .. zeek:see:: reading_live_traffic
+## .. zeek:see:: reading_live_traffic packet_sources
 function reading_traces%(%): bool
 	%{
 	return val_mgr->GetBool(reading_traces);
 	%}
 
+## Returns: a list of packet sources being read by Zeek.
+##
+## .. zeek:seek:: reading_live_traffic reading_traces
+function packet_sources%(%): PacketSourceList
+	%{
+	auto ps_type = internal_type("PacketSource")->AsRecordType();
+	auto psl_type = internal_type("PacketSourceList")->AsVectorType();
+	auto rval = make_intrusive<VectorVal>(psl_type);
```

Comment:
Is there any real reason to use `intrusive_ptr` here if you're going to just immediately detach it?

Reply:
Not especially for this code, but thought it good to get in the habit of using them since:

* it does the right thing in case an exception is thrown.  e.g. we do already have existing problems with raw pointers leaking memory due to non-fatal interpreter exceptions.
* similar to above, but seen a couple mistakes made in the past when someone goes to insert new logic that returns from a function early and miss Unref'ing a raw pointer
* expect detaches like this eventually go away as code gradually refactors to use `intrusive_ptr`